kbase-extension/static/kbase/js/common/ui.js

```diff
@@ -335,120 +312,197 @@ define([
 
-    function showInfoDialog(arg) {
-        const dialog = renderInfoDialog(arg.title, arg.body, arg.okLabel || 'OK'),
-            dialogId = html.genId(),
-            confirmNode = document.createElement('div');
-        let kbaseNode, modalNode;
+    /**
+     * Show a generic, configurable dialog
+     * @param {Object} arg with keys
+     *      {string}    title
+     *      {string}    body
+     *      {string}    type - the type of dialog; 'confirm' dialogs can return true
+     *                  or false, whereas others return false
+     *      {string}    includeOK (opt) - if defined, an 'OK' button is added with
+     *                  the value as the label
+     *      {string}    includeCancel (opt) - if defined, a 'Cancel' button is added
+     *                  with the value as the label
+     *      {string}    bsClass (opt) - the bootstrap theme to use for the modal
+     *                  one of 'primary', 'success', 'info', 'warning', 'danger'
+     *
+     * @returns {Promise} that resolves to false if the modal is dismissed using the
+     * cancel button, the close button, by clicking on the modal backdrop, or by pressing
+     * the escape key. It resolves to true if the ok button is clicked or the user hits
+     * enter.
+     */
+    function showGenericDialog(args) {
+        const dialog = _renderModal(args),
+            confirmNode = _generateConfirmNode(dialog),
+            modalDialogNode = _setUpModalNodes(confirmNode);
 
-        confirmNode.id = dialogId;
-        confirmNode.innerHTML = dialog;
+        // this shows the modal
+        $(modalDialogNode).modal({ keyboard: false });
 
-        // top level element for kbase usage
-        kbaseNode = document.querySelector('[data-element="kbase"]');
-        if (!kbaseNode) {
-            kbaseNode = document.createElement('div');
-            kbaseNode.setAttribute('data-element', 'kbase');
-            document.body.appendChild(kbaseNode);
-        }
+        let resolution = false;
 
-        // a node upon which to place Bootstrap modals.
-        modalNode = kbaseNode.querySelector('[data-element="modal"]');
-        if (!modalNode) {
-            modalNode = document.createElement('div');
-            modalNode.setAttribute('data-element', 'modal');
-            kbaseNode.appendChild(modalNode);
+        if (args.type === 'confirm') {
+            modalDialogNode.querySelector('[data-element="ok"]').addEventListener('click', () => {
+                resolution = true;
+            });
         }
 
-        modalNode.appendChild(confirmNode);
+        modalDialogNode.addEventListener('keyup', (e) => {
+            // 13 = enter, 27 = escape
+            // the key is sometimes a string, hence == instead of ===
+            if (e.key == 13 || e.key == 27) {
+                $(modalDialogNode).modal('hide');
+                if (e.key == 13 && args.type === 'confirm') {
+                    resolution = true;
+                }
+            }
```

Comment:
One button modal: pressing escape or enter dismisses the modal
Two button modal: pressing escape is equivalent to clicking 'cancel'
pressing enter is equivalent to clicking 'ok'
(this is documented in the modal constructor below)